crates/floresta-wire/src/p2p_wire/chain_selector.rs

```diff
@@ -325,8 +325,17 @@ where
                 // if they disagree on this current block, we need to look in the previous one
                 height -= 1;
             }
+            hash = self.chain.get_block_hash(height).unwrap();
```

Comment:
This is one more issue I found in `find_who_is_lying` function, when we move forward in blockchain to find the fork, we also have to change the `BlockHash` along with the height. Previously it was only incrementing the `height`.